### PR TITLE
feat: 🎸 [JIRA: HCPSDKFIORIUIKIT-1940] profile header

### PR DESCRIPTION
**Pull request by xiaoyu0722**
|static|animatable|
|-|-|
|Simulator Screenshot - iPad Pro 13-inch (M4) (16GB) - 2024-07-01 at 15 04 04|Simulator Screenshot - iPad Pro 13-inch (M4) (16GB) - 2024-07-01 at 15 04 09|

### CODE REVIEW

File: Apps/Examples/Examples/FioriSwiftUICore/ProfileHeader/ProfileHeaderAnimatableExample.swift [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+                })
+                .frameReader(in: .named("profileHeader"), rect: { rect in
+                    if rect.origin.y >= 0 {
+                        v = 0
+                    } else if rect.origin.y <= -45 {
+                        v = 1
+                    } else {
+                        v = -rect.origin.y / 45
+                    }
+                })
```
Comment (billzhou0223) [inline Apps/Examples/Examples/FioriSwiftUICore/ProfileHeader/ProfileHeaderAnimatableExample.swift:60]: This should be handled by SDK if possible.

**Review by billzhou0223 — CHANGES_REQUESTED** [review]

File: Sources/FioriSwiftUICore/_FioriStyles/ProfileHeaderStyle.fiori.swift [hunk 2]
```diff
@@ -0,0 +1,216 @@
+import FioriThemeManager
+import Foundation
+import SwiftUI
+
+// Base Layout style
+public struct ProfileHeaderBaseStyle: ProfileHeaderStyle {
```
Comment (dyongxu) [inline Sources/FioriSwiftUICore/_FioriStyles/ProfileHeaderStyle.fiori.swift:6]: 1. Add support for "right to left" layout
2. Add accessibility support for the component

**Review by dyongxu — CHANGES_REQUESTED** [review]
"right to left" layout to be supported
accessibility voice over support to be added

File: Sources/FioriSwiftUICore/_ComponentProtocols/CompositeComponentProtocols.swift [hunk 3]
```diff
@@ -279,3 +279,15 @@ protocol _BannerMessageComponent: _TitleComponent, _CloseActionComponent, _TopDi
     /// The action to be performed when the banner is tapped.
     var bannerTapAction: (() -> Void)? { get }
 }
+
+// sourcery: CompositeComponent
+protocol _ProfileHeaderComponent: _DetailImageComponent, _TitleComponent, _SubtitleComponent, _DescriptionComponent {
+    // sourcery: @Binding
+    // sourcery: defaultValue = .constant(0)
+    /// A binding float for transition of profile header. Value should be in range of 0 to 1. Default value is 0.
+    var transitionProgress: CGFloat { get }
```
[the remark this answers is not in the document]
Comment (xiaoyu0722) [reply Sources/FioriSwiftUICore/_ComponentProtocols/CompositeComponentProtocols.swift:288]: Like [swiftui-introspect](https://github.com/siteline/swiftui-introspect) to get the offset of its superview to calculate the progress?

File: Sources/FioriSwiftUICore/_FioriStyles/ProfileHeaderStyle.fiori.swift (hunk 2 above)
Comment (xiaoyu0722) [reply Sources/FioriSwiftUICore/_FioriStyles/ProfileHeaderStyle.fiori.swift:6]: RTL is supported, and accessibility is also enabled by default.

**Review by dyongxu — APPROVED** [review]

**Comment by dyongxu** [comment]
@xiaoyu0722 Looks like some other PR has checked in some generated files, causing your merge conflict. Need to pull latest  main to resolve conflict.

**Comment by xiaoyu0722** [comment]
Hi @billzhou0223 , need review again.

**Review by billzhou0223 — APPROVED** [review]

**Pull request merged**